lint.go

```diff
@@ -1038,22 +1038,38 @@ func (f *file) lintRanges() {
 	})
 }
 
-// lintErrorf examines errors.New calls. It complains if its only argument is an fmt.Sprintf invocation.
+// lintErrorf examines errors.New and testing.Error calls. It complains if their only arguments are fmt.Sprintf invocations.
 func (f *file) lintErrorf() {
+	testingTName := ""
 	f.walk(func(node ast.Node) bool {
-		ce, ok := node.(*ast.CallExpr)
-		if !ok {
-			return true
-		}
-		if !isPkgDot(ce.Fun, "errors", "New") || len(ce.Args) != 1 {
-			return true
-		}
-		arg := ce.Args[0]
-		ce, ok = arg.(*ast.CallExpr)
-		if !ok || !isPkgDot(ce.Fun, "fmt", "Sprintf") {
-			return true
+		switch ce := node.(type) {
+		case *ast.FuncDecl:
+			name := ce.Name.Name
+			if strings.HasPrefix(name, "Test") {
+				for _, field := range ce.Type.Params.List {
+					if f.render(field.Type) == "*testing.T" && len(field.Names) > 0 {
+						testingTName = field.Names[0].Name
+					}
+				}
+			}
+
+		case *ast.CallExpr:
+			isErrorsNew := isPkgDot(ce.Fun, "errors", "New")
+			isTestingError := isPkgDot(ce.Fun, testingTName, "Error")
+			if !(isErrorsNew || isTestingError) || len(ce.Args) != 1 {
+				return true
+			}
+			arg := ce.Args[0]
+			ce, ok := arg.(*ast.CallExpr)
+			if !ok || !isPkgDot(ce.Fun, "fmt", "Sprintf") {
+				return true
+			}
+			err := "should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)"
```

Comment:
don't duplicate this whole string just to change one piece. Use f.render on the right bit of the AST (probably a SelectorExpr).